### ISSUE TITLE
Support for role and purpose in network module

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-network#339]
### TL;DR

Only subnets-beta supports configuring purpose and role, but this seems to have been promoted to a non-beta feature, so we can add support to the top-level module

### Terraform Resources

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_subnetwork

### Additional information

I think the proposed change consists of modifying subnets, and potentially removing the now-unnecessary subnets-beta module (since all fields are supported by the main subnets module). This change consists of adding the inputs:

* purpose
* role

for the subnets module: https://github.com/terraform-google-modules/terraform-google-network/blob/a7b91bc1a70c323b99eecc229a70b449b6bda401/modules/subnets/main.tf#L28-L33

**Comment by jawnsy** [comment]
I think this is still a valid problem

**Comment by bharathkkb** [comment]
@jawnsy I believe we added this via https://github.com/terraform-google-modules/terraform-google-network/pull/343
